site/content/api/Credentials-API.md

```diff
@@ -0,0 +1,316 @@
+
+## Remove Credentials
+
+Clients use this command to *remove* credentials registered for a device. Once the credentials are removed, the device may no longer be able to authenticate with protocol adapters using the authentication mechanism that the removed credentials corresponded to.
+
+**Message Flow**
+
+*TODO* add sequence diagram
+
+**Request Message Format**
+
+The following table provides an overview of the properties a client needs to set on a *remove credentials* message in addition to the [Standard Request Properties]({{< relref "#standard-request-properties" >}}).
+
+| Name             | Mandatory | Location                 | Type           | Description |
+| :--------------- | :-------: | :----------------------- | :------------- | :---------- |
+| *device_id*      | yes       | *application-properties* | UTF-8 *string* | MUST contain the ID of the device. |
+| *action*         | yes       | *application-properties* | UTF-8 *string* | MUST contain the value `update`. |
+
+The body of the message MUST consist of a single *AMQP Value* section containing a UTF-8 encoded string representation of a single JSON object as follows:
+
+~~~json
+{
+  "type": "${credential_type}",
+  "auth-id": "${authentication_identifier}"
+}
+~~~
+
+The `type` property MUST contain the exact type of credentials to remove. The `auth-id` property MUST contain the authentication identifier to remove the credentials for.
+
+**Response Message Format**
+
+A response to a *remove credentials* request only contains the [Standard Response Properties]({{< relref "#standard-response-properties" >}}).
+
+The response message's body is empty.
+
+The response message's *status* property may contain the following codes:
+
+| Code  | Description |
+| :---- | :---------- |
+| *204* | No Content, the credentials have been removed successfully. |
+| *404* | Not Found, there are no credentials registered matching the critera. |
+
+For status codes indicating an error (codes in the `400 - 499` range) the message body MAY contain a detailed description of the error that occurred.
+
+# Standard Message Properties
+
+Due to the nature of the request/response message pattern of the operations of the Credentials API, there are some standard properties shared by all of the request and response messages exchanged as part of the operations.
+
+## Standard Request Properties
+
+The following table provides an overview of the properties shared by all request messages regardless of the particular operation being invoked.
+
+| Name             | Mandatory | Location                 | Type           | Description |
+| :--------------- | :-------: | :----------------------- | :------------- | :---------- |
+| *action*         | yes       | *application-properties* | UTF-8 *string* | MUST be set to the value defined by the particular operation being invoked. |
```

Comment:
I like it :-) We should then, however, also change it in the Registration API as well to use it consistently.